src/services/leafletWatchHelpers.js

```diff
@@ -20,7 +20,17 @@ angular.module("leaflet-directive")
   to set watching to once and its watch depth.
   */
   var _maybeWatch = function(scope, thingToWatchStr, watchOptions, initCb){
-      return _maybe(scope, '$watch', thingToWatchStr, watchOptions, initCb);
+      var localWatchOptions, watchMethod;
```

Comment:
Also either `.type` or `.methodName` should be explicitly `'$watchCollection'` and `'$watch'` it is easier this way and we can handle whatever other scope things that might come up without needing to worry about prepending or whatever.